src/groups/GroupCard.test.tsx

```diff
@@ -0,0 +1,106 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+import { writeText } from 'clipboard-polyfill/text';
+import React from 'react';
+import { GroupCard } from 'src/groups/GroupCard';
+import { CurrentUserGroupMembership } from 'src/libs/ajax/Groups';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+
+jest.mock('src/libs/nav', (): typeof import('src/libs/nav') => ({
+  ...jest.requireActual('src/libs/nav'),
+  getLink: jest.fn((link) => link),
+}));
+
+jest.mock('clipboard-polyfill/text', () => ({
+  ...jest.requireActual('clipboard-polyfill/text'),
+  writeText: jest.fn().mockResolvedValue(true),
+}));
+
+describe('GroupCard', () => {
+  const testGroup1: CurrentUserGroupMembership = {
+    groupEmail: 'group1@email.com',
+    groupName: 'test-group1-name',
+    role: 'member',
+  };
+  const testGroup2: CurrentUserGroupMembership = {
+    groupEmail: 'group2@email.com',
+    groupName: 'test-group2-name',
+    role: 'admin',
+  };
+
+  it('displays the group name, email, and role', async () => {
+    // Arrange
+    const { getByText } = render(<GroupCard group={testGroup1} onDelete={jest.fn()} onLeave={jest.fn()} />);
+    // Act
+    // Assert
+    expect(getByText(testGroup1.groupName, { exact: false })).toBeDefined();
+    expect(getByText(testGroup1.groupEmail, { exact: false })).toBeDefined();
+    expect(getByText(testGroup1.role, { exact: false })).toBeDefined();
+  });
+
+  it('calls the leave callback when button to leave the group is clicked', async () => {
+    // Arrange
+    const mockLeaveCallback = jest.fn();
+    const { getByText, getByRole } = render(
+      <GroupCard group={testGroup1} onDelete={jest.fn()} onLeave={mockLeaveCallback} />
+    );
+    expect(getByText(testGroup1.groupName, { exact: false })).toBeDefined();
+
+    // Act
+    const menu = getByRole('button', { name: `Action Menu for Group: ${testGroup1.groupName}` });
+    fireEvent.click(menu);
```

Comment:
There has been a move towards using `userEvent` to interact with UI elements in unittests. TBH, I don't recall what the advantages are-- @msilva-broad, do you know?

Reply:
I do think that one thing about the userEvent style is that if you await `user.click`, you don't have to use `waitFor` to test the expected behavior afterwards because it guarantees to have updated.